src/project-tests/bar-chart-tests.js

```diff
@@ -1,196 +1,274 @@
+          assert.isNotNull(
+            bar.getAttribute('data-gdp'),
+            'Could not find property "data-gdp" in bar '
+          );
+        });
+      });
+
+      reqNum++;
+      it(`${reqNum}. The bar elements' "data-date" properties should match the 
+      order of the provided data`,
+      function(done) {
+        $.getJSON(
+          'https://raw.githubusercontent.com/FreeCodeCamp/' +
+          'ProjectReferenceData/master/GDP-data.json',
+          function(res) {
+            try {
+              const bars = document.querySelectorAll('rect.bar');
+              assert.isAtLeast(
+                bars.length,
+                1,
+                'no elements with the class of "bar" are detected '
               );
+              bars.forEach(function(bar, i) {
+                var currentBarDate = bar.getAttribute('data-date');
+                assert.equal(
+                  currentBarDate,
+                  res.data[i][0],
+                  'Bars should have date data in the same order as the ' +
+                  'provided data '
+                );
+              });
+              done();
+            } catch (e) {
+              done(e);
             }
-            done();
-          } catch (e) {
-            done(e);
           }
-        }
-      );
-    });
+        );
+      });
 
-    reqNum++;
-    it(`${reqNum}. The "data-gdp" properties should match the order of the
-    provided data`,
-    function(done) {
-      $.getJSON(
-        'https://raw.githubusercontent.com/FreeCodeCamp/ProjectReferenceData/' +
-        'master/GDP-data.json',
-        function(res) {
-          try {
-            const bars = document.getElementsByClassName('bar');
-            assert.isAtLeast(
-              bars.length,
-              1,
-              'no elements with the class of "bar" are detected '
-            );
-            for (var i = 0; i < bars.length; i++) {
-              var currentBarGdp = bars[i].getAttribute('data-gdp');
-              assert.equal(
-                currentBarGdp,
-                res.data[i][1],
-                'Bars should have gdp data in the same order as the ' +
-                'provided data '
+      reqNum++;
+      it(`${reqNum}. The bar elements' "data-gdp" properties should match the 
+      order of the provided data`,
+      function(done) {
+        $.getJSON(
+          'https://raw.githubusercontent.com/FreeCodeCamp/' +
+          'ProjectReferenceData/master/GDP-data.json',
+          function(res) {
+            try {
+              const bars = document.querySelectorAll('rect.bar');
+              assert.isAtLeast(
+                bars.length,
+                1,
+                'no elements with the class of "bar" are detected '
               );
+              bars.forEach(function(bar, i) {
+                var currentBarGdp = bar.getAttribute('data-gdp');
+                assert.equal(
+                  currentBarGdp,
+                  res.data[i][1],
```

Comment:
Nice. I didn't realize we needed to use an index here, but we can still get away with using `forEach`.